File: KNN/KNN.php
<?php
$data = file_get_contents("https://datahub.io/machine-learning/iris/r/iris.csv");
$rows = explode("\n",$data);
$s = array();
foreach($rows as $row) {
    $s[] = str_getcsv($row);
}
$csvData = array_slice($s,1,100);
preProcessing($csvData);



function preProcessing($csvData){
shuffle($csvData);
$train_data = array_slice($csvData,0,50);
$test_data = array_slice($csvData,50,50);

$train_labels = array();

for ($i = 0; $i < count($train_data); $i++) {
  $train_labels[$i] = $train_data[$i][4];
}

KNN($train_data, $test_data[0], 25);
}

function distance($train,$test) {
  $dist = 0;
  for ($i=0; $i < count($train); $i++) {
    $dist = $dist + pow(($train[$i] - $test[$i]), 2);
  }
  return sqrt($dist);
}

function KNN($train_sample,$test,$N) {

  $dist_array = array();
  for ($i=0; $i < count($train_sample); $i++) {
    $dist_array[$i][1] = distance($train_sample[$i], $test);
    $dist_array[$i][0] = $i;
  }

  $labels = array();
  for ($i = 0; $i < $N; $i++) {
    $labels[$i] = $train_sample[$dist_array[$i][0]][4];

  }

  $arr_freq = array_count_values($labels);
  arsort($arr_freq);
  echo array_keys($arr_freq)[0];

}
 ?>

File: KNN/KNN.py
import csv
import urllib2
import random

def distance(train,test):
    dist =0
    for value in train:
        dist+=pow((value - $test[$i]), 2)
def KNN(train_data,test_data,N):
    pass


def preProcessing(results):
    random.shuffle(results)
    train_data = results[0:50]
    test_data = results[50:100]

    train_labels = []
    for value in train_data:
        train_labels.append(value[4])

    KNN(train_data,test_data[0], 25);


url = 'https://datahub.io/machine-learning/iris/r/iris.csv'
response = urllib2.urlopen(url)
rows = csv.reader(response)
results = []
for row in rows:
    results.append(row)
results = results[1:101]
preProcessing(results)


#
# import csv
#
# results = []
# with urllib2.urlopen("https://datahub.io/machine-learning/iris/r/iris.csv") as csvfile:
#     reader = csv.reader(csvfile, quoting=csv.QUOTE_NONNUMERIC) # change contents to floats
#     for row in reader: # each row is a list
#         results.append(row)
#
# print(results)
